streams/src/main/scala/io/confluent/examples/streams/MapFunctionScalaExample.scala

```diff
@@ -0,0 +1,73 @@
+package io.confluent.examples.streams
+
+import java.util.Properties
+
+import io.confluent.examples.streams.utils.SystemTimestampExtractor
+import org.apache.kafka.common.serialization.{ByteArrayDeserializer, ByteArraySerializer, StringDeserializer, StringSerializer}
+import org.apache.kafka.streams.kstream.{KStream, KStreamBuilder, KeyValue}
+import org.apache.kafka.streams.{KafkaStreaming, StreamingConfig}
+
+/**
+  * Demonstrates how to perform simple, state-less transformations via map functions.
+  *
+  * Use cases include e.g. basic data sanitization, data anonymization by obfuscating sensitive data
+  * fields (such as personally identifiable information aka PII).
+  *
+  * Requires Scala 2.12 for Java 8 and Java lambda support.
+  */
+class MapFunctionScalaExample {
+
+  def main(args: Array[String]) {
+    val builder: KStreamBuilder = new KStreamBuilder
+
+    // Read the input Kafka topic into a KStream instance.
+    val textLines: KStream[Array[Byte], String] = builder.stream(new ByteArrayDeserializer, new StringDeserializer, "TextLinesTopic")
+
+    // Variant 1: using `mapValues`
+    val uppercasedWithMapValues: KStream[Array[Byte], String] = textLines.mapValues(_.toUpperCase())
+
+    // Write (i.e. persist) the results to a new Kafka topic called "UppercasedTextLinesTopic".
+    //
+    // Alternatively you could also explicitly set the serializers instead of relying on the default
+    // serializers specified in the streaming configuration (see further down below):
+    //
+    //     uppercasedWithMapValues.to("UppercasedTextLinesTopic"), new ByteArraySerializer, new StringSerializer)
+    //
+    uppercasedWithMapValues.to("UppercasedTextLinesTopic")
+
+    // Variant 2: using `map`, modify value only (equivalent to variant 1)
+    val uppercasedWithMap: KStream[Array[Byte], String] = textLines.map((key, value) => new KeyValue(key, value.toUpperCase()))
```

Comment:
I added an implicit conversion from Scala's `Tuple2` to Streams' `KeyValue` in an upcoming PR.